### PR TITLE
0.20.1 Adding test_037 storage managing switch.

### PR DESCRIPTION
**Pull request by antip00**

### CODE REVIEW

File: tests/e2e/steps.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from testflows.connect import Shell
+
+
+@TestStep(Given)
+def get_shell(self, timeout=300):
+    """Create shell terminal."""
+    try:
+        shell = Shell()
+        shell.timeout = timeout
+        yield shell
```
Comment (Slach) [inline tests/e2e/steps.py:11]: are you sure finally block will executed properly ?

File: tests/e2e/test_operator.py [hunk 2]
```diff
@@ -2907,6 +2908,83 @@ def test_035(self):
             self.context.shell.close()
 
 
+@TestScenario
+@Requirements(RQ_SRS_026_ClickHouseOperator_Managing_StorageManagementSwitch("1.0"))
+@Name("test_037. storageManagement switch")
+def test_037(self):
```
Comment (Slach) [inline tests/e2e/test_operator.py:2914]: test scenario LGTM

File: tests/e2e/steps.py (hunk 1 above)
Comment (antip00) [reply tests/e2e/steps.py:11]: Yes, it is common syntax, it calls final block in the end of scenario.

File: tests/e2e/manifests/chi/test-037-1-storagemanagement-switch.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+apiVersion: "clickhouse.altinity.com/v1"
+kind: "ClickHouseInstallation"
+metadata:
+  name: "test-037-storagemanagement-switch"
+spec:
+  useTemplates:
+    - name: clickhouse-version
+  configuration:
+    zookeeper:
```
Comment (alex-zaitsev) [inline tests/e2e/manifests/chi/test-037-1-storagemanagement-switch.yaml:9]: Remove ZooKeeper and convert to a single replica test to make test faster. You are testing 0-0-0 node anyway.

File: tests/e2e/manifests/chi/test-037-1-storagemanagement-switch.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  templates:
+    volumeClaimTemplates:
+      - name: disk1
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 1Gi
+    podTemplates:
```
Comment (alex-zaitsev) [inline tests/e2e/manifests/chi/test-037-1-storagemanagement-switch.yaml:32]: If you are using clickhouse-version template in defaults above, you do not need podTemplate at all, it is automatically created from template

**Review by alex-zaitsev — CHANGES_REQUESTED** [review]

**Pull request merged**